code/datums/components/tackle.dm

```diff
@@ -0,0 +1,448 @@
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.stamina_cost = stamina_cost
+	src.base_knockdown = base_knockdown
+	src.range = range
+	src.speed = speed
+	src.skill_mod = skill_mod
+	src.min_distance = min_distance
+
+	var/mob/P = parent
+	to_chat(P, "<span class='notice'>You are now able to launch tackles! You can do so by activating throw intent, and clicking on your target with an empty hand.</span>")
+
+	addtimer(CALLBACK(src, .proc/resetTackle), base_knockdown, TIMER_STOPPABLE)
+
+/datum/component/tackler/Destroy()
+	var/mob/P = parent
+	to_chat(P, "<span class='notice'>You can no longer tackle.</span>")
+	..()
+
+/datum/component/tackler/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_CLICKON, .proc/checkTackle)
+	RegisterSignal(parent, COMSIG_MOVABLE_IMPACT, .proc/sack)
+	RegisterSignal(parent, COMSIG_MOVABLE_POST_THROW, .proc/registerTackle)
+
+/datum/component/tackler/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_MOB_CLICKON, COMSIG_MOVABLE_IMPACT))
+
+///Store the thrownthing datum for later use
+/datum/component/tackler/proc/registerTackle(mob/living/carbon/user, datum/thrownthing/TT)
+	tackle = TT
+
+///See if we can tackle or not. If we can, leap!
+/datum/component/tackler/proc/checkTackle(mob/living/carbon/user, atom/A, params)
+	if(!user.in_throw_mode || user.get_active_held_item() || user.pulling)
+		return
+
+	if(user.lying)
+		to_chat(user, "<span class='warning'>You must be standing to tackle!</span>")
+		return
+
+	if(tackling)
+		to_chat(user, "<span class='warning'>You're not ready to tackle!</span>")
+		return
+
+	if(user.has_movespeed_modifier(MOVESPEED_ID_SHOVE)) // can't tackle if you just got shoved
+		to_chat(user, "<span class='warning'>You're too off balance to tackle!</span>")
+		return
+
+	user.face_atom(A)
+
+	var/list/modifiers = params2list(params)
+	if(modifiers["alt"] || modifiers["shift"] || modifiers["ctrl"] || modifiers["middle"])
+		return
+
+	tackling = TRUE
+	RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/checkObstacle)
+
+	if(can_see(user, A, 7))
+		user.visible_message("<span class='warning'>[user] leaps at [A]!</span>", "<span class='danger'>You leap at [A]!</span>")
+	else
+		user.visible_message("<span class='warning'>[user] leaps!</span>", "<span class='danger'>You leap!</span>")
+
+	if(get_dist(user, A) < min_distance)
+		A = get_ranged_target_turf(user, get_dir(user, A), min_distance) //TODO: this only works in cardinals/diagonals, make it work with in-betweens too!
+
+	user.Knockdown(base_knockdown, TRUE, TRUE)
+	user.adjustStaminaLoss(stamina_cost)
+	user.throw_at(A, range, speed, user, FALSE)
+	addtimer(CALLBACK(src, .proc/resetTackle), base_knockdown, TIMER_STOPPABLE)
+	return(COMSIG_MOB_CANCEL_CLICKON)
+
+/*
+	sack() is called when you actually smack into something, assuming we're mid-tackle. First it deals with smacking into non-carbons, in two cases:
```

Comment:
first line is the title, break it from the rest with an empty `*` line otherwise it'll stick the whole paragraph in the table of contents